### DISCUSSION TITLE
How to Dynamically Set app.baseURL in Nuxt 3 with Environment Variables After Build

### DISCUSSION THREAD

**Discussion by a20246** [nuxt/nuxt#25839] [Questions]
Hello everyone,

I hope this message finds you well. I'm currently working on a Nuxt 3 project and encountered a challenge that I hope to get some insights on from this knowledgeable community.

After running npm run build, I've been able to successfully override settings in runtimeConfig.public using environment variables prefixed with NUXT_PUBLIC_XXXX. This part works as expected, and I'm able to dynamically change these settings based on my environment variables.

However, I'm facing an issue with app.baseURL. Despite setting it in my nuxt.config with an environment variable, it doesn't seem to reflect the changes after the build process. Here is the relevant part of my nuxt.config for reference:

```
export default defineNuxtConfig({
    runtimeConfig: {
      public: {
        hostname: process.env.HOSTNAME,
        prefix: process.env.PREFIX,
      },
    },
    app: {
      baseURL: process.env.PREFIX,
    },
});
```
I need to be able to dynamically change the app.baseURL based on different environment variables after npm run build. Is there any recommended approach or workaround that could allow me to achieve this? Any advice or suggestions would be greatly appreciated.

If you think this approach might not be suitable, or if you have any thoughts or alternative suggestions, I'm very open to discussion. Your feedback is highly valued and can help guide me to a better solution.

Thank you very much for your time and help.

**Comment by xjccc** [comment]
change `NUXT_APP_BASE_URL` have a try?

[Alexander Lichter](https://www.youtube.com/@TheAlexLichter) 
this video will help you
[Nuxt's runtimeConfig - The most common mistake](https://www.youtube.com/watch?v=_FYV5WfiWvs)
